.github/workflows/ci-goreleaser-core.yaml

```diff
@@ -0,0 +1,69 @@
+name: Continuous Integration - GoReleaser - Core
```

Comment:
I did some similar work in this repo, with reusable workflows: https://github.com/jpkrohling/otelcol-distributions

Basically, I defined the generic part here: https://github.com/jpkrohling/otelcol-distributions/blob/main/.github/workflows/base-release.yaml

And then each distribution would just pass parameters:
- https://github.com/jpkrohling/otelcol-distributions/blob/main/.github/workflows/release-loadbalancing.yaml
- https://github.com/jpkrohling/otelcol-distributions/blob/main/.github/workflows/release-sidecar.yaml
- https://github.com/jpkrohling/otelcol-distributions/blob/main/.github/workflows/release-tracing.yaml

Reply:
@jpkrohling perfect, I can implement this reusable pattern now instead of later

Reply:
Done